sfdx-project.json/sfdx-project.schema.json

```diff
@@ -84,41 +109,111 @@
       "description": "By default, the OAuth port is 1717. However, change this port if this port is already in use, and you plan to create a connected app in your Dev Hub org to support JWT-based authorization."
     },
     "plugins": {
+      "title": "CLI Plugins custom settings",
       "type": "object",
-      "description": "Salesforce CLI plugins and configuration used with this project."
+      "description": "Salesforce CLI plugin configurations used with this project.",
+      "additionalProperties": true
     },
     "packageAliases": {
+      "title": "Aliases for packaging ids",
       "type": "object",
-      "description": "The Salesforce CLI updates this file with the aliases when you create a package or package version. You can also manually update this section for existing packages or package versions. You can use the alias instead of the cryptic package ID when running CLI force:package commands."
+      "description": "The Salesforce CLI updates this file with the aliases when you create a package or package version. You can also manually update this section for existing packages or package versions. You can use the alias instead of the cryptic package ID when running CLI force:package commands.",
+      "patternProperties": {
+        "^[ -~]+$": {
+          "type": "string",
+          "pattern": "^(0Ho|033|05i|04t)([a-zA-Z0-9]{15}|[a-zA-Z0-9]{12})$"
+        }
+      }
     }
   },
   "definitions": {
+    "features": {
+      "type": "array",
+      "title": "Features",
+      "description": "The list of features names to enable",
+      "oneOf": [
+        {
+          "type": "string",
+          "title": "Features",
+          "description": "The list of feature names to enable as a string"
+        },
+        {
+          "type": "array",
+          "title": "Features",
+          "description": "The list of feature names to enable as an array",
+          "items": {
+            "type": "string",
+            "title": "Feature Name",
+            "description": "Name of the feature to enable"
+          }
+        }
+      ]
+    },
+    "orgPreferences": {
```

Comment:
I'm pretty sure this has been fully deprecated and now should only be specified through the referenced definition file.  How is deprecation being handled?